### DISCUSSION TITLE
MQTT notification

### DISCUSSION THREAD

**Discussion by rikardronnkvist** [containrrr/watchtower#676] [Feature Requests]
Would really like to see notification via MQTT...

Really easy to work with and you can connect other automations based on the MQTT messages.

**Comment by piksel** [comment]
Please make a request for it on [containrrr/shoutrrr](https://github.com/containrrr/shoutrrr).

If I were to implement this I would probably also need to know a bit about how you would expect it to work.

**Reply by rikardronnkvist** [reply]
Hm... well thinking some more about this Im not really sure if it is a "Notification" that would fit in Shoutrrr or if it should be a separate module.

MQTT is a lightweight publish/subscribe protocol that transports json-messages on a servicebus.
So in Watchtower I would like to see "all kind of messages" published to the servicebus instead of having just log-messages.

For example, say that you would get
Topic: /watchtower/container/mycontainername/stopped
Message {somestuff:"bla bla bla"}

Then you can subscribe to topic's like /watchtower/container/*/stopped and start automations based on messages on that servicebus... :)

**Comment by simonaronsson** [comment]
This is a really interesting feature! I think however it might require some more discussion to nail down whether this would fit into notifications or should be treated separately.

In the interest of setting reasonable expectations, I don't think @piksel nor I will get the opportunity to work on this in the near future, so if anyone feels up to it, feel free to elaborate on a design proposal.

**Comment by dougle03** [comment]
There are standard ways to implement MQTT messaging. Most that want to use it will already have an MQTT broker; publishing log messages to MQTT should really be an add on where all types and content of log entries are by default published, then anyone who wants to listen (Subscribe) to these will be able to. Receiving MQTT is a little more complex as the actions on received messages require a little more thought... either way, implementation would be awesome, especially if scheduling of updates can be better controlled. More important now that Docker has fully implemented pull limits...

**Comment by rikardronnkvist** [comment]
Didn't even think about Watchtower recieving messages... but that would be really cool.

Why would you like to control the flow via MQTT..?
Might be that you prefer to update some images at a specific time and some other based on another event (CI/CD-pipeline for instance).
Example: You can set up a webhook from GitHub, recieve that hook with some automation tool and then send a MQTT-message that would trigger the Watchtower-update...

**Comment by simonaronsson** [comment]
Locking this discussion in favor of https://github.com/containrrr/shoutrrr/issues/86
